kod/object/passive/ornobj.kod

```diff
@@ -735,14 +750,101 @@ messages:
    {
       local f;
 
-      f = LOOK_NO;
+      f = 0;
+      if NOT pbLook { f = f | LOOK_NO; }
       if pbHanging { f = f | OF_HANGING; }
       if pbFlicker { f = f | FLICKERING_YES; }
       if pbNoMoveOn { f = f | MOVEON_NO; }
       
       return f;
    }
 
+   ShowDesc()
+   {
+      if psCustomDesc <> $
+      {
+         AddPacket(4, oo_custom_template_rsc, 4, psCustomDesc);
+
+         return;
+      }
+
+      propagate;
+   }
+
+   PlaceAt(what=$)
+   "Admin supported\n"
+   "Moves this ornamental object as close to <what>'s current room and location as possible."
+   {
+      local lPos, oRoom;
+
+      if what <> $
+      {
+         lPos = Send(what, @GetPos);
+         oRoom = Send(SYS, @UtilGetRoom, #what=what);
+         if lPos <> $ AND oRoom <> $
+         {
+            if oRoom <> poOwner
+            {
+               Send(oRoom, @NewHold, #what=self,
+                  #new_row=First(lPos), #new_col=Nth(lPos, 2),
+                  #fine_row=Nth(lPos, 3), #fine_col=Nth(lPos, 4),
+                  #new_angle=Send(what, @GetAngle));
+            }
+            else
+            {
+               Send(SYS, @UtilGoNearSquare, #what=self, #where=oRoom,
+                  #new_row=First(lPos), #new_col=Nth(lPos, 2),
+                  #fine_row=Nth(lPos, 3), #fine_col=Nth(lPos, 4),
+                  #new_angle=Send(what, @GetAngle));
+            }
+         }
+      }
+
+      return;
+   }
+
+   SetCustomDesc(text=$)
```

Comment:
Given that you found that dynamic strings are never garbage collected, do you think this is safe?  I'm not sure how often it will be used.

Reply:
I'm pretty confident in this, actually.  It's the same technique used by ballots, books, and various other items that allow users to write in them, all of which get used with more frequency than this probably will due to it being admin-only.  The testing I did showed that strings do seem to be garbage collected after a save once they were no longer associated with something.  These were my test steps, in case you're curious:

- Set custom description for 2 ornamental objects.
- Save to garbage collect/compact references.
- Check and note both string ids (string 5439 and 5440, as well as 5438 which is unrelated).
- Remove custom description for the object using 5439.
- Save again.
- Note that string 5440 has moved down to 5439, there is nothing in 5440, and 5438 is what it was unchanged.  So the original 5439 was garbage collected successfully due to no longer being used.